receiver/awscontainerinsightreceiver/internal/gpu/decorator.go

```diff
@@ -0,0 +1,204 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gpu
```

Comment:
Can we make this package generic , I get it now its only for GPU but if we make it configurable i.e it adds unit based on the Map provided, also adding the K8 Label seem to re-usable say in Neuron monitor.

Reply:
Strongly agree with this.